chainer/links/connection/n_step_rnn.py

```diff
@@ -121,36 +120,59 @@ def __call__(self, hx, xs, **kwargs):
                 Each element ``xs[i]`` is a :class:`chainer.Variable` holding
                 a sequence.
         """
+        (hy,), ys = self._call([hx], xs, **kwargs)
+        return hy, ys
+
+    def _call(self, hs, xs, **kwargs):
+        """Calls RNN function.
+
+        Args:
+            hs (list of ~chainer.Variable or None): Lisit of hidden states.
+                Its length depends on its implementation.
+                If ``None`` is specified zero-vector is used.
+            xs (list of ~chainer.Variable): List of input sequences.
+                Each element ``xs[i]`` is a :class:`chainer.Variable` holding
+                a sequence.
+
+        Returns:
+            tuple: hs
+        """
         argument.check_unexpected_kwargs(
             kwargs, train='train argument is not supported anymore. '
             'Use chainer.using_config')
         argument.assert_kwargs_empty(kwargs)
 
         assert isinstance(xs, (list, tuple))
-        xp = cuda.get_array_module(hx, *xs)
+        xp = cuda.get_array_module(*(list(hs) + list(xs)))
         indices = argsort_list_descent(xs)
         indices_array = xp.array(indices)
 
         xs = permutate_list(xs, indices, inv=False)
-        if hx is None:
-            hx = self.init_hx(xs)
-        else:
-            hx = permutate.permutate(hx, indices_array, axis=1, inv=False)
+        hxs = []
+        for hx in hs:
+            if hx is None:
+                hx = self.init_hx(xs)
+            else:
+                hx = permutate.permutate(hx, indices_array, axis=1, inv=False)
+            hxs.append(hx)
 
         trans_x = transpose_sequence.transpose_sequence(xs)
 
-        ws = [[w.w0, w.w1] for w in self]
-        bs = [[w.b0, w.b1] for w in self]
+        ws = [[getattr(w, 'w%d' % i) for i in range(self.n_weights)]
+              for w in self]
+        bs = [[getattr(w, 'b%d' % i) for i in range(self.n_weights)]
+              for w in self]
 
-        hy, trans_y = self.rnn(
-            self.n_layers, self.dropout, hx, ws, bs, trans_x,
-            activation=self.activation)
+        args = [self.n_layers, self.dropout] + hxs + [ws, bs, trans_x]
+        result = self.__class__.rnn(*args)
```

Comment:
Why do you use `__class__` ?

Reply:
I don't know why, but without it, it does not work. I'll try to fix it.